tests/cases/compiler/declarationEmitLateBoundAssignments2.ts

```diff
@@ -0,0 +1,19 @@
+// @strict: true
+// @declaration: true
+// @target: es6
+
+// repro from https://github.com/microsoft/TypeScript/issues/54811
+
+const c = "C"
+
+export function decl () {}
+decl["B"] = 'foo'
```

Comment:
what does the BinaryExpression case look like? If there's not a test for it here or somewhere else, please add one.

Reply:
It's the `decl2` case. This is the baseline diff without the `isBinaryExpression` check:

<details>
<summary>baseline diff</summary>

```diff
diff --git a/tests/baselines/reference/declarationEmitLateBoundAssignments.js b/tests/baselines/reference/declarationEmitLateBoundAssignments.js
index b76db45754..763511ec0d 100644
--- a/tests/baselines/reference/declarationEmitLateBoundAssignments.js
+++ b/tests/baselines/reference/declarationEmitLateBoundAssignments.js
@@ -38,5 +38,4 @@ const a = foo[dashStrMem];
 export declare function foo(): void;
 export declare namespace foo {
     var bar: number;
-    var strMemName: string;
 }
diff --git a/tests/baselines/reference/declarationEmitLateBoundAssignments2.js b/tests/baselines/reference/declarationEmitLateBoundAssignments2.js
index 4ec90c2606..b2d3efc2a8 100644
--- a/tests/baselines/reference/declarationEmitLateBoundAssignments2.js
+++ b/tests/baselines/reference/declarationEmitLateBoundAssignments2.js
@@ -37,9 +37,6 @@ export declare namespace decl {
     var B: string;
 }
 export declare function decl2(): void;
-export declare namespace decl2 {
-    var C: number;
-}
 export declare const arrow: {
     (): void;
     B: string;
```
</details>